AdSecGH/Parameters/AdSecSection.cs

```diff
@@ -85,36 +82,90 @@ public AdSecSection(
       }
 
       LocalPlane = local;
-      CreatePreview(ref _profile, ref _profileEdge, ref _profileVoidEdges, ref _profileColour, ref _rebars,
-        ref _rebarEdges, ref _linkEdges, ref _rebarColours, ref _subProfiles, ref _subEdges, ref _subVoidEdges,
-        ref _subColours);
+      AssignPreviewData();
     }
 
+    internal AdSecSection() { }
+
     public IDesignCode DesignCode { get; set; }
-    public bool IsValid => Section != null;
+    public bool IsValid => SolidBrep != null && SolidBrep.IsValid;
     public Plane LocalPlane { get; set; }
     public ISection Section { get; set; }
     internal Brep SolidBrep => _profile;
-    internal List<Brep> SubBreps => _subProfiles;
 
     public AdSecSection Duplicate() {
-      if (this == null) {
-        return null;
-      }
-
-      var dup = (AdSecSection)MemberwiseClone();
-      return dup;
+      return IsValid ? (AdSecSection)MemberwiseClone() : null;
     }
 
     public override string ToString() {
       return Section.Profile.Description();
     }
 
-    internal void CreatePreview(
-      ref Brep profile, ref Polyline profileEdge, ref List<Polyline> profileVoidEdges,
-      ref DisplayMaterial profileColour, ref List<Brep> rebars, ref List<Circle> rebarEdges, ref List<Curve> linkEdges,
-      ref List<DisplayMaterial> rebarColours, ref List<Brep> subProfiles, ref List<Polyline> subEdges,
-      ref List<List<Polyline>> subVoidEdges, ref List<DisplayMaterial> subColours, IPoint offset = null) {
+    private void GenerateLocalPlanePreviewAxes() {
+      if (LocalPlane == Plane.WorldXY || LocalPlane == Plane.WorldYZ || LocalPlane == Plane.WorldZX) {
+        return;
+      }
+
+      var area = Section.Profile.Area();
+      double pythogoras = Math.Sqrt(area.As(AreaUnit.SquareMeter));
```

Comment:
I will agree with AI that this is misspelled. But perhaps we should name it `sideSize`?